core/src/create.jl

```diff
@@ -225,13 +210,44 @@ end
 
 function FlowBoundary(db::DB, config::Config)::FlowBoundary
     static = load_structvector(db, config, FlowBoundaryStaticV1)
-    defaults = (; active = true)
-    static_parsed = parse_static(static, db, "FlowBoundary", defaults)
-    return FlowBoundary(
-        static_parsed.node_id,
-        static_parsed.active,
-        static_parsed.flow_rate,
-    )
+    time = load_structvector(db, config, FlowBoundaryTimeV1)
+
+    static = load_structvector(db, config, FlowBoundaryStaticV1)
+    time = load_structvector(db, config, FlowBoundaryTimeV1)
+
+    static_node_ids = Set(static.node_id)
+    time_node_ids = Set(time.node_id)
+    msg = "FlowBoundary cannot be in both static and time tables"
+    @assert isdisjoint(static_node_ids, time_node_ids) msg
+    node_ids = get_ids(db, "FlowBoundary")
+
+    msg = "FlowBoundary node IDs don't match"
+    @assert issetequal(node_ids, union(static_node_ids, time_node_ids)) msg
+
+    active = BitVector()
+    flow_rate = Float64[]
+
+    for node_id in node_ids
+        if node_id in static_node_ids
+            static_idx = findfirst(static.node_id .== node_id)
```

Comment:
This line is correct but can be improved in two ways. `a .== b` creates an intermediate boolean vector that we can avoid, e.g. by `findfirst(==(node_id), static.node_id)`. `==(node_id)` creates a function that takes one argument that checks if it is equal to `node_id`, similar to `x -> x == node_id`.

But we also know from `load_structvector` that `static.node_id` is sorted, so we can use binary search to speed things up, which will help if there are many FlowBoundaries.

```suggestion
            static_idx = searchsortedfirst(static.node_id, node_id)
```

Reply:
Neat, I found some more instances of this